content/blog/build-a-twitter-bot-using-nodejs/index.md

```diff
@@ -0,0 +1,103 @@
+
+> `package.json` holds all the dependencies of a project.
+
+Since our project will be dependent on **twitter** client library, it's time we install this dependency in our project. Run `npm install twitter` to include this in your `package.json` .
+
+### 3. Setup Files
+
+Now we will start with the actual coding part. At the moment, your project folder will have a `node_modules` folder, a `package.json` file and a `package-lock.json` file. If you have gone with the default options while running `npm init`, then `index.js` is the file we will be working upon.
+
+Open `index.js`. To be able to use the **twitter** library which we have installed, you need to include it in `index.js`. 
+
+`var Twitter = require('twitter');`
+
+Also, remember the **keys** and **tokens** you got from Twitter Developer Portal? It's time we use them. 
+
+```js
+var client = new Twitter({
+	consumer_key: process.env.CONSUMER_KEY,
+	consumer_secret: process.env.CONSUMER_SECRET,
+	access_token_key: process.env.ACCESS_TOKEN_KEY,
+	access_token_secret: process.env.ACCESS_TOKEN_SECRET
+});
+```
+
+**Do not add your Consumer or Access Token keys or secrets directly in the `index.js`.** These should be managed accordingly using *enviroment variables*.
+
+> We will use `dotenv` library to manage our enviroment variables in NodeJS.  Run `npm install dotenv` and follow [these](https://www.npmjs.com/package/dotenv#usage) instructions. Your `.env` will look something like this.
+
+```js
+CONSUMER_KEY=........................................
+CONSUMER_SECRET=.....................................
+ACCESS_TOKEN_KEY=....................................
+ACCESS_TOKEN_SECRET=.................................
+```
+
+### 4. Start Coding 🎯
+
+I will use [Official Joke API](https://github.com/15Dkatz/official_joke_api) to automatically fetch content for my tweet. You can visit the repository and learn more about the endpoints. But the only [endpoint](https://github.com/15Dkatz/official_joke_api#grab-a-random-joke) we will work with is [random_joke](https://official-joke-api.appspot.com/random_joke). Click on this and you will receive a JSON response with fields id, type, setup and punchline. The fields we are concerned with are setup and punchline. 
+
+We will use [axios](https://github.com/axios/axios) for calling the API endpoint. Run `npm install axios` and add `var axios = require('axios');` in `index.js`.
+
+Now everything has been laid out. We need to connect the dots. The Official Joke API will return a JSON response. We will grab the required data i.e. the setup and the punchline from it. And then make use of [client.post](https://github.com/desmondmorris/node-twitter/tree/master/examples#tweet) functionality of **twitter** package.
+
+```js
+axios("https://official-joke-api.appspot.com/random_joke").then(Response => {
```

Comment:
@Ninjavin Does axios support api request like this? Shouldn't this be either a `.get` or a config object`?